nixos/modules/services/matrix/synapse.nix

```diff
@@ -937,6 +936,13 @@ in {
 
   config = mkIf cfg.enable {
     assertions = [
+      {
+        assertion = lib.any (listenerSupportsResource "client") cfg.settings.listeners;
```

Comment:
Is this really required? I admittedly don't know the entire Client-Server API but would it be possible to have a worker handle all C-S API requests?

Reply:
Ohh, you're right :facepalm: 

Good point, thanks! Do you think it'd make sense to look for a `client` listener in both workers and main listeners then? Right now we don't support multi-machine deployments anyways.

Reply:
Looking for a client listener in either `cfg.settings.listeners` or `cfg.workers.*.worker_listeners` should be enough as we already have the assertion for a replication listener (and now with this PR for `cfg.settings.instance_map.main` as well).

Reply:
So it just occurred to me that this will also fix `registerNewMatrixUser` when using workers (and not serving `client` from the main process anymore).

Fixed, thanks!